scheduler/src/cook/mesos.clj

```diff
@@ -75,9 +78,90 @@
 
 (counters/defcounter [cook-mesos mesos mesos-leader])
 
+(defn make-mesos-driver
+  "Creates a mesos driver
+   
+   Parameters:
+   mesos-framework-name     -- string, name to use when connecting to mesos. 
+                               Will be appended with version of cook
+   mesos-role               -- string, (optional) The role to connect to mesos with
+   mesos-principal          -- string, (optional) principal to connect to meso with
+   gpu-enabled?             -- boolean, (optional) whether cook will schedule gpu jobs
+   mesos-failover-timeout   -- long, (optional) time in milliseconds mesos will wait 
+                               for framework to reconnect. 
+                               See http://mesos.apache.org/documentation/latest/high-availability-framework-guide/
+                               and search for failover_timeout
+   mesos-master             -- str, (optional) connection string for mesos masters
+   scheduler                -- mesos scheduler implementation
+   framework-id             -- str, (optional) Id of framework if it has connected to mesos before
+   "
+  ([config scheduler]
+   (make-mesos-driver config scheduler nil))
+  ([{:keys [mesos-framework-name mesos-role mesos-principal gpu-enabled? 
+            mesos-failover-timeout mesos-master] :as config}
+    scheduler framework-id]
+   (apply mesomatic.scheduler/scheduler-driver
+          scheduler
+          (merge
+            {:user ""
+             :name (str mesos-framework-name
+                        "-"
+                        cook.util/version)
+             :checkpoint true}
+            (when mesos-role
+              {:role mesos-role})
+            (when mesos-principal
+              {:principal mesos-principal})
+            (when gpu-enabled?
+              {:capabilities [{:type :framework-capability-gpu-resources}]})
+            (when mesos-failover-timeout
+              {:failover-timeout mesos-failover-timeout})
+            (when framework-id
+              {:id {:value framework-id}}))
+          mesos-master
+          (when mesos-principal
+            [{:principal mesos-principal}]))))
+
+(defn get-mesos-utilization
+  [mesos-master-hosts]
+  (let [mesos-master-urls (map #(str "http://" % ":5050/metrics/snapshot") mesos-master-hosts)
```

Comment:
Yes but I copied this from cook.mesos.scheduler as is. Let's make it configurable in a different PR